### PR TITLE
FOGL-1442: addition of getChildCategories and addChildCategory

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-1442: addition of getChildCategories and addChildCategory

Exceptions refactoring

### CODE REVIEW

File: C/services/core/configuration_manager.cpp [hunk 1]
```diff
@@ -56,53 +57,79 @@ ConfigurationManager* ConfigurationManager::getInstance(const string& host,
...
-		throw StorageOperationException();
-	}
+		// Query via Storage client
+		allCategories = m_storage->queryTable("configuration", qAllCategories);
+		if (!allCategories)
+		{
+			// Data layer error
+			throw CategoryDetailsEx();
+		}
+		if (!allCategories->rowCount())
```
Comment (MarkRiddoch) [inline C/services/core/configuration_manager.cpp]: You could combine you tests if you wanted to make the code more concise

File: C/services/core/configuration_manager.cpp [hunk 2]
```diff
@@ -123,50 +151,82 @@ ConfigCategory ConfigurationManager::getCategoryAllItems(const string& categoryN
...
-	// Get first row
-	ResultSet::RowIterator it = categoryItems->firstRow();
-	ResultSet::Row* row = *it;
-	if (!row)
-	{
-		delete categoryItems;
-		throw CategoryDetailsException();
-	}	
+		// Cayegory not found
+		if (!categoryItems->rowCount())
```
Comment (MarkRiddoch) [inline C/services/core/configuration_manager.cpp]: Again you could combine these two tests

File: C/services/core/configuration_manager.cpp (hunk 1 above)
Comment (pintomax) [reply C/services/core/configuration_manager.cpp]: yes, done

File: C/services/core/configuration_manager.cpp (hunk 2 above)
Comment (pintomax) [reply C/services/core/configuration_manager.cpp]: The code here returns two different exceptions and we cannot combine it

**Pull request closed without merging**